source/Coop.Core/Server/Services/Clans/Handler/ServerClanAdoptionHandler.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using Common.Network;
+using Coop.Core.Client.Services.Clans.Messages;
+using Coop.Core.Server.Services.Clans.Messages;
+using GameInterface.Services.Clans.Messages;
+using Serilog;
+using System;
+
+namespace Coop.Core.Server.Services.Clans.Handler
+{
+    /// <summary>
+    /// Handles all changes to clans on server.
+    /// </summary>
+    public class ServerClanAdoptionHandler : IHandler
```

Comment:
The server should also have functionality to detect HeroAdopted. Create a new handler method for HeroAdopted and all other events.

This new handler will detect and internal hero adopted and should do the same as `Handle(MessagePayload<NetworkAdoptHeroRequest> obj)`